### DISCUSSION TITLE
Code mountpoint

### DISCUSSION THREAD

**Discussion by kptdobe** [adobe/helix-home#341] [Ideas]
A recurrent customer need is to have a central repo with the code (template) and then various repos for each site implementing the design. Our response is to use Github actions to keep the repo in sync. We are currently prototyping a GH action machinery that will also be able to handle the case where one of the implementing site has changed something (PR or override).
This leads to a lot of code duplication and a lot of CO2 consume to run the GH actions to sync everything.

What if in the fstab you could add a `code` property pointing to a GH repo ? 

To create an instance of your template, you would simply need one fstab file. That's it. 
If you create the same file in the instance repo, it wins and we have an override mechanism.

Since the code is managed by the bot, it would be easy on code change to keep the folders in S3 up-to-date. And the task for the bot is only to find if there is an override or not.

@trieloff @tripodsan @davidnuescheler @rofe WDYT ?

**Comment by tripodsan** [comment]
if you want to duplicate the files in the code-bus, the template repository needs to know all the users. so whenever files are updated in the template, you also need to copy them to all the _forks_.

but we could do this maybe similar to the content-blueprints, having a code-blueprints....

**Reply by kptdobe** [reply]
> if you want to duplicate the files in the code-bus, the template repository needs to know all the users. so whenever files are updated in the template, you also need to copy them to all the forks.

Yes, that's the idea. But nothing complicated, just need to maintain a list of dependencies.

> but we could do this maybe similar to the content-blueprints, having a code-blueprints....

What is the content-blueprints ?

**Reply by tripodsan** [reply]
> What is the content-blueprints ?

https://www.hlx.live/docs/configuration#blueprints
